backends-common/elasticsearch/src/test/java/org/apache/james/backends/es/ClientProviderImplTest.java

```diff
@@ -19,19 +19,15 @@
 
 package org.apache.james.backends.es;
 
-import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
-import java.util.Optional;
-
-import org.elasticsearch.common.settings.Settings;
 import org.junit.Test;
 
 public class ClientProviderImplTest {
 
     @Test
     public void fromConfigurationShouldThrowOnNull() {
-        assertThatThrownBy(() -> ClientProviderImpl.fromConfiguration(null))
+        assertThatThrownBy(() -> new ClientProvider(null))
```

Comment:
why not ClientProvider.fromConfiguration() ?

Reply:
because now it's supposed to be injected. The test name is wrong, I'll rename it.

Reply:
my question was more about why not using a static named method instead of a constructor

Reply:
and I answered: there's no point having a factory method when you intend to inject the class